### PR TITLE
Project Sync transitioned to new success/error handling

### PR DESCRIPTION
**Pull request by febbraro**
This was the big daddy, but I think I got it.  Required a lil bit of refactoring to make it all work.

### CODE REVIEW

File: cli/commands/project_sync.go [hunk 1]
```diff
@@ -97,56 +97,60 @@ func (cmd *ProjectSync) RunStart(ctx *cli.Context) error {
...
-		case "linux":
-			cmd.out.Verbose.Printf("Setting up local volume: %s", volumeName)
-			cmd.SetupBindVolume(volumeName, workingDir)
-		default:
-			cmd.out.Verbose.Printf("Starting sync with volume: %s", volumeName)
-			cmd.StartUnisonSync(ctx, volumeName, cmd.Config, workingDir)
-		}
-	} else {
-		cmd.out.Error.Fatal(err)
+	var workingDir, err = cmd.DeriveLocalSyncPath(cmd.Config, ctx.String("dir"))
```
Comment (grayside) [inline cli/commands/project_sync.go]: Why does this work without the `:=`? When I've tried things like this before, I'd have err yelling at me.

File: cli/commands/project_sync.go [hunk 2]
```diff
@@ -97,56 +97,60 @@ func (cmd *ProjectSync) RunStart(ctx *cli.Context) error {
...
+	if err != nil {
+		return cmd.Error(err.Error(), "SYNC-PATH-ERROR", 12)
```
Comment (grayside) [inline cli/commands/project_sync.go]: Hm, maybe we should define cmd.EXTERNAL_ERROR_CODE and so on for more clarity about the numbers? I realize that's outside immediate scope but reading this out of the context of the other PR this immediately jumps out at me as "why is this a number 12?"

File: cli/commands/project_sync.go (hunk 1 above)
Comment (febbraro) [reply cli/commands/project_sync.go]: := is just a shorthand for not needing to `var` it. The var and = I believe is redefining it.

File: cli/commands/project_sync.go (hunk 2 above)
Comment (febbraro) [reply cli/commands/project_sync.go]: I think some constants for standard error codes are probably a good idea. We can tackle that in another PR

**Pull request merged**